File: sdks/ts/src/sdk/models/operations/getsignrequests.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetSignRequestsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=marker" })
  marker?: string;
}


export class GetSignRequestsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetSignRequestsQueryParams;
}


export class GetSignRequestsResponseOutput extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  signRequests?: shared.SignRequestsOutput;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getsignrequests.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetSignRequestsQueryParams struct {
	Limit  *int64  `queryParam:"style=form,explode=true,name=limit"`
	Marker *string `queryParam:"style=form,explode=true,name=marker"`
}

type GetSignRequestsRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetSignRequestsQueryParams
}

type GetSignRequestsResponseOutput struct {
	ClientError  *shared.ClientError
	ContentType  string
	SignRequests *shared.SignRequestsOutput
	StatusCode   int64
}
